api/users/permissions.py

```diff
@@ -13,3 +13,16 @@ def has_object_permission(self, request, view, obj):
             return True
         else:
             return obj == request_user
+
+
+class ReadOnlyOrCurrentUserRelationship(permissions.BasePermission):
+    """ Check to see if the request is coming from the currently logged in user,
+    and allow non-safe actions if so.
+    """
+    def has_object_permission(self, request, view, obj):
+        assert isinstance(obj, dict)
+        request_user = request.user
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        else:
+            return obj['self'] == request_user
```

Comment:
`==` can sometimes be unpredictable when comparing ODM objects. It's safer to compare PKs ( e.g. `request_user._id`)